suite-common/walletconnect/src/walletConnectMiddleware.ts

```diff
@@ -0,0 +1,40 @@
+import { createMiddlewareWithExtraDeps } from '@suite-common/redux-utils';
+import { accountsActions } from '@suite-common/wallet-core';
+
+import { walletConnectActions } from './walletConnectActions';
+import * as walletConnectThunks from './walletConnectThunks';
+
+export const prepareWalletConnectMiddleware = createMiddlewareWithExtraDeps(
+    async (action, { dispatch, next, extra }) => {
+        await next(action);
+
+        if (accountsActions.updateSelectedAccount.match(action) && action.payload.account) {
+            dispatch(
+                walletConnectThunks.switchSelectedAccountThunk({
```

Comment:
not sure how but I got this here 
![image](https://github.com/user-attachments/assets/5d5b65cd-317c-44e7-8630-ed27be0ec1f0)